### DISCUSSION TITLE
Request and Import AEV from other nodes when `InvalidBlockStateRootHashException` occurs

### DISCUSSION THREAD

**Discussion by area363** [planetarium/libplanet#2190] [Ideas]
When `InvalidBlockStateRootHashException` occurs when executing a block, the node cannot move past that block.

In this case, the general way to fix this issue is to:
1) Download a new snapshot where the tip is greater than the problematic block.
2) Manually fix Li9c action codes and resolve `InvalidBlockStateRootHashException`.

Unfortunately, both methods take a long time to execute.

Would it be possible to request AEV from other nodes and use that as the block evaluation result and move on to the next block?

**Comment by area363** [comment]
Related Issue: https://github.com/planetarium/lib9c/issues/1237

**Comment by moreal** [comment]
How about providing all evaluations of the block through `InvalidBlockStateRootHashException`? They may be too big but useful to debug.

**Comment by moreal** [comment]
In the Ethereum implementation, it called `TxExecution` (a.k.a `txResult`) as `Receipt`. It represents the transaction's execution result and it contains the intermediate state root hash. I think the feature will be useful if provided.

ref:
- https://cs.github.com/ethereum/go-ethereum/blob/3b2a6b34d92b84f6bfab058a46d9fc73c21365c7/core/state_processor.go#L95
- https://cs.github.com/ethereum/go-ethereum/blob/3b2a6b34d92b84f6bfab058a46d9fc73c21365c7/core/state/statedb.go#L826

To achieve it, it should apply transaction one by one instead of `GetTotalDelta`.
